docs/tutorials/custom_segmentation_trainer.ipynb

```diff
@@ -0,0 +1,514 @@
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 9278198/9278198 [00:00<00:00, 21183125.27it/s]\n",
+      "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2,3]\n",
+      "\n",
+      "  | Name          | Type             | Params | Mode \n",
+      "-----------------------------------------------------------\n",
+      "0 | model         | Unet             | 32.5 M | train\n",
+      "1 | criterion     | CrossEntropyLoss | 0      | train\n",
+      "2 | train_metrics | MetricCollection | 0      | train\n",
+      "3 | val_metrics   | MetricCollection | 0      | train\n",
+      "4 | test_metrics  | MetricCollection | 0      | train\n",
+      "-----------------------------------------------------------\n",
+      "32.5 M    Trainable params\n",
+      "0         Non-trainable params\n",
+      "32.5 M    Total params\n",
+      "130.087   Total estimated model params size (MB)\n",
+      "242       Modules in train mode\n",
+      "0         Modules in eval mode\n",
+      "/opt/conda/envs/torchgeo/lib/python3.12/site-packages/lightning/pytorch/loops/fit_loop.py:298: The number of training batches (1) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "47c8ef6ff2f147a287fcd137058b2509",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Training: |          | 0/? [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "db9116f3e41245c6897b885aa322dcbc",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Validation: |          | 0/? [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "`Trainer.fit` stopped: `max_epochs=1` reached.\n"
+     ]
+    }
+   ],
+   "source": [
+    "trainer.fit(task, dm)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test model\n",
+    "\n",
+    "Finally, we test the model (optionally loading from a previously saved checkpoint)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# You can load directly from a saved checkpoint with `.load_from_checkpoint(...)`\n",
+    "# Note that you can also just call `trainer.test(task, dm)` if you've already trained\n",
+    "# the model in the current notebook session.\n",
+    "\n",
+    "task = CustomSemanticSegmentationTask.load_from_checkpoint(\n",
```

Comment:
I know this is like the whole purpose of #2317, but also completely unnecessary to demonstrate a custom trainer. If you remove this, we could backport this to 0.6.1 so it makes it into the stable docs immediately.

Reply:
I'm not in a rush to get this into the stable docs, and think it is a nice use case.